kaitai/kaitaistream.cpp

```diff
@@ -691,9 +694,143 @@ std::string kaitai::kstream::bytes_to_str(std::string src, std::string src_enc)
     return dst;
 }
 #elif defined(KS_STR_ENCODING_NONE)
-std::string kaitai::kstream::bytes_to_str(std::string src, std::string src_enc) {
+std::string kaitai::kstream::bytes_to_str(const std::string src, const char* src_enc) {
     return src;
 }
+#elif defined(KS_STR_ENCODING_WIN32API)
+#include <windows.h>
+#include <limits>
+
+// Unbreak std::numeric_limits<T>::max, as otherwise MSVC substitutes "useful" max() macro.
+#undef max
```

Comment:
What a mess 🤦‍♂️

Given the circumstances, I guess this workaround is OK, though.

Reply:
It is :( I'm yet to see a non-trivial Windows C++ project which doesn't do a bunch of tweaks like that (e.g. undefining min/max, defining _UNICODE, etc), sadly.